src/noit_socket_listener.c

```diff
@@ -254,12 +245,16 @@ listener_metric_track_or_log(void *vrxc, const char *name,
       }
     }
   }
-  noit_stats_mark_metric_logged(noit_check_get_stats_inprogress(rxc->check), m, mtev_false);
-  pthread_mutex_lock(&rxc->flushlock);
-  mtev_boolean inserted = mtev_hash_store(rxc->immediate_metrics, m->metric_name, strlen(m->metric_name), m);
```

Comment:
I think this changes the logic in a way that will cause data loss.

Reply:
how?  If it's not in ``logged`` how can it be in ``immediate_metrics``?

Reply:
I was worried about it being logged and then showing up immediately again where it would find itself already logged and skip the immediate insert.